taca_ngi_pipeline/deliver/deliver_grus.py

```diff
@@ -203,42 +207,89 @@ def deliver_project(self):
              logger.error("Not delivering becouse wrong mover version detected")
              return False
         # moved this part from constructor, as we can create an object without running the delivery (e.g. to check_delivery_status)
-        #check if the project directory already exists, if so abort
+        if self.delivery_name != None:
+            self.stagingpathhard = os.path.join(os.path.split(self.stagingpathhard)[0], self.delivery_name)
         hard_stagepath = self.expand_path(self.stagingpathhard)
+        soft_stagepath = self.expand_path(self.stagingpath)
+        all_projects = {p:GrusProjectDeliverer(p) for p in (self.projectid,) + self.include_project}
```

Comment:
hmm.. this doesn't feel sound.. at least you shouldn't have to create a new instance for `self.projectid`, right? I think I would be more comfortable if the functionality you are needing the `GrusProjectDeliverer` instances for was implemented outside of the `GrusProjectDeliverer` class itself. But I don't have a good overview of this class so I can't say for sure.

Reply:
`GrusProjectDeliverer` class is very similar to `ProjectDeliverer`, in the sense it uses `self` class methods/variables to do stuff (example `self.get_delivery_status()`, `self.stagingpathhard`, etc). So why I also include main project (`self.projectid`) is cause when a we include projects, the exact same things (test, stageing, etc) are going to be done for those projects in the exact same manner as the main project.

So in order to avoid redundant code I just instantiate the class again for all projects and iterate over it.